ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Abstract.hs

```diff
@@ -57,6 +58,13 @@ class ( SupportedBlock blk
                    -> LedgerState blk
                    -> Except (LedgerError blk) (LedgerState blk)
 
+  -- | Re-apply a block to the very same ledger state it was applied in before.
+  reapplyLedgerBlock :: HasCallStack
```

Comment:
Is the TODO above about issue #596 done now, or is that still relevant?

Reply:
Oh and why does this have a `HasCallStack` when `applyLedgerBlock` does not? Should they both have it or neither? Oh, is it because `reapplyLedgerBlock` can still fail but only as a programmer error, and if so does that mean we're still checking too much? :-)

Reply:
I think we could have a slightly more extensive commentary here about what reapply is for, and what pre-condition it has, and thus what kinds of checks we allow the ledger layer to skip based on it. There's probably some good commentary we can copy from a related ticket or slack discussion.

We can perhaps refer to the docs for `BlockValidationMode` and `TxValidationMode` from the ledger lib.

Reply:
I'm unsure about the #596 TODO comment. Apparently #848 should have solved #596, but that PR actually introduced this TODO comment. Will ask @ruhatch about this.

Reply:
- Ru has confirmed that the TODO comment can be removed.
- Added further doc to `reapplyLedgerBlock`